### ISSUE TITLE
Cannot force front/back-channel logout with id_token_hint when "Remember Me" is off

### ISSUE THREAD

**Issue by mig5** [ory/hydra#3132]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

Hi,

The page https://www.ory.sh/docs/hydra/debug/logout and diagram at https://www.ory.sh/docs/hydra/concepts/logout says that if "Remember Me" is not true, one can still trigger front/back-channel logout by passing `id_token_hint` in the logout request.

This does not seem to be true.

I logged in to an RP without 'Remember Me' set, and saved the id_token somewhere in the RP (as an attribute of the RP's session).

I then issued a logout in my RP, which sent to the OP with the `id_token_hint` containing that ID token, as https://op-example.com/oauth2/sessions/logout?id_token_hint=xxxxxxxxxxxxxxxxxxxxx&post_logout_redirect_uri=https%3A%2F%2Fmy-example-rp.com

I expected to see Hydra go to the Login&Consent app's /logout endpoint with the logout_challenge and perform front/back-channel logout, but instead it went straight to the post_logout_redirect_uri.

### Reproducing the bug

1. Log into an RP app but don't set remember_me in the login request.
2. Ensure your RP captures the id_token somehow
3. Log out of the RP, ensuring that its 'logout' route that invokes the logout request at OP, sends the id token as the id_token_hint param, and your post logout redirect URI param
4. Expect to see in your browser that Hydra hits the 'Logout' endpoint to do front/back channel logout, despite the lack of an OP session, because id_token_hint was sent in the request
5. Find that it didn't, and just went to the post logout redirect URI

### Version

v1.11.7

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Docker

**Comment by mig5** [comment]
To be clear - if I *do* set 'Remember me' when logging in, then indeed because the Hydra OP session exists, front/back channel takes place on logout.

But the docs explicitly state that even if there is no 'Remember me', the presence of id_token_hint is enough to still invoke front/channel logout. This does not seem to be the case.

**Comment by mig5** [comment]
Sorry, I haven't tried again since - my organization made a separate decision, they wanted to 'force' Remember Me to always be 'on', so we just avoided the problem.

**Comment by alnr** [comment]
Indeed, the behavior is wrong for OAuth2 logouts with remember=false. This needs to be fixed in code; the only workaround right now is what @mig5 said [here](https://github.com/ory/hydra/issues/3132#issuecomment-1470872860).

**Comment by aeneasr** [comment]
This can probably be implemented now because the ID token hint includes the sid / session id claim, which was missing before and iirc was the reason why we needed a session cookie to figure out which session to invalidate

**Comment by fkammer** [comment]
I dug into this and the root cause is much smaller than it looks: login sessions **are** persisted even when the login request is accepted with `remember: false` — `verifyAuthentication` calls `ConfirmLoginSession` for every non-skipped login, only the session cookie is withheld. So the row (including subject and the `sid` that ends up in the ID token) exists in `hydra_oauth2_authentication_session` with `remember = FALSE`.

The only thing breaking logout is the `Where("remember = TRUE")` filter in `GetRememberedLoginSession`:
https://github.com/ory/hydra/blob/4174065ffb052799890f7480f5360a877a67ffc1/persistence/sql/persister_consent.go#L234

Both logout entry points go through it and treat the resulting `ErrNotFound` as "session already gone":
- `issueLogoutVerifier` (RP-initiated logout with `id_token_hint`) silently redirects to `post_logout_redirect_uri` without any front-/back-channel logout: https://github.com/ory/hydra/blob/4174065ffb052799890f7480f5360a877a67ffc1/consent/strategy_default.go#L942-L947
- `HandleHeadlessLogout` (`DELETE /admin/oauth2/auth/sessions/login?sid=...`) becomes a silent 204 no-op: https://github.com/ory/hydra/blob/4174065ffb052799890f7480f5360a877a67ffc1/consent/strategy_default.go#L1099-L1105

Everything downstream is already remember-agnostic (`ListClientsWithLogoutURLsForSubjectAndSID` queries flows by `login_session_id` + subject, `DeleteLoginSession` has no remember filter), which confirms @aeneasr's suggestion: the `sid` from the `id_token_hint` is all that's needed.

**Proposed fix:** add a second method to the `LoginManager` interface (e.g. `GetLoginSession`), identical to `GetRememberedLoginSession` minus the remember filter, and use it at the two logout call sites above. The filtered variant stays in place for the cookie-based SSO lookup in `authenticationSession`, so `remember=false` sessions still never enable SSO skip. No schema or API changes needed — the admin endpoint would finally behave as its docs already promise. Plus tests: note that the existing "headless logout with remember=false" test only covers consent-remember=false; the test helper hardcodes login `Remember: true`, which is why this was never caught.

Happy to open a PR for this — let me know if you'd take it.
